.changes/unreleased/Features-20241029-181728.yaml

```diff
@@ -0,0 +1,6 @@
+kind: Features
+body: Emit debug logging event whenever artifacts are written
```

Comment:
Do we still need this?

Reply:
I think "need" is a strong word. We wanted to add this event because we couldn't tell if the writing of the artifact was actually happening when debugging the issue of #10934 in cloud. Having this log could thus reduce debug time in the future if something like this happens again. Additionally, it's possible that throughout execution we end up writing an artifact multiple times, which is likely wasteful. Having a log like this would be helpful in identifying if that is happening. Additionally, the fix for the #10934 isn't guaranteed by the fix in [0223f3a](https://github.com/dbt-labs/dbt-core/pull/10936/commits/0223f3a16ea0344bcc56ebc8ce4ef867164587a1). That fix is fairly narrowly scoped. It is possible if an exception gets raised elsewhere, we might run into a situation again where an artifact is not being written (thus the log is useful).

---
Tangentially, I just admitted the fix in [0223f3a](https://github.com/dbt-labs/dbt-core/pull/10936/commits/0223f3a16ea0344bcc56ebc8ce4ef867164587a1) is pretty limited. This is unfortunately necessary. I tried moving the exception handling up some levels, but quickly you run into a labrynth of catching too much in instances where failure is expected 🫠